packages/components/src/components/pin-code/pin-code-utils.ts

```diff
@@ -0,0 +1,48 @@
+import type { FormState } from '../../utils/form/form-state';
+import { removeAttribute, setAttribute } from '../../utils';
+
+export const PIN_CODE_TYPES = ['number', 'password'] as const;
+export type PinCodeType = (typeof PIN_CODE_TYPES)[number];
+
+export type PinCodeUpdateEvent = { value: string | number };
+
+export type PinCodeState = FormState;
+
+export const isTypeNumber = (type: string): boolean => {
+  return type === 'number';
+};
+
+export const inputIsSingleDigit = (input: string): boolean => input.length === 1 && /\d/.test(input);
+
+export const joinInputValues = (pinCodeElements: HTMLInputElement[]): string =>
+  pinCodeElements.map((el) => el.value).join('');
+
+export const initHiddenInput = (
+  host: HTMLElement,
+  value: string,
+  disabled: boolean,
+  required: boolean
+): HTMLInputElement => {
+  const hiddenInput = document.createElement('input');
+  setAttribute(hiddenInput, 'name', 'hidden-input');
```

Comment:
This will cause duplicate names when 2 components are used within the same form. I think the consumer needs to be able to set this via property on the component.